content/1_docs/2_cookbook/9_setup/0_understanding-oop/recipe.txt

```diff
@@ -0,0 +1,468 @@
+
+If you want to follow along with this recipe, you need the following:
+
+- A web server (ideally on your (link: docs/cookbook/setup/development-environment text: local machine))
+- A current version of PHP
+- A Kirby installation ((link: docs/guide/quickstart text: Starterkit or Plainkit))
+- A code/text editor of your choice
+
+## What is OOP?
+
+Object Oriented Programming is a programming paradigm based on the concept of objects that have certain properties and methods that allow interacting with these objects. The properties and methods of such an object are described in a class, which acts as the model or blueprint for such objects, whereas an object is an instance of this class, i.e. a specific embodiment of the class.
+
+In a larger software program, you typically have multiple classes whose objects can interact with each other.
+
+### A simple class example
+
+Let's examine a very simple example to get a basic idea. Here we define a class called `Book` with some properties that should apply to all objects of this class, and methods to work with objects of this class.
+
+```php "book.php"
+<?php
+class Book {
+
+    private $title;
+    private $author;
+    private $isbn;
+    private $price;
+    private $publisher;
+
+    public function __construct( string $title, string $author) {
+        $this->title  = $title;
+        $this->author = $author;
+    }
+
+    public function getTitle(): string {
+        return $this->title;
+    }
+
+    public function getAuthor(): string {
+        return $this->author;
+    }
+
+    public function getPrice(): float {
+        return $this->price;
+    }
+
+    public function getPublisher(): string {
+        return $this->publisher;
+    }
+    
+    public function getIsbn(): string {
+        return $this->isbn;
+    }
+    
+    public function setTitle(string $title): void {
+        $this->title = $title;;
+    }
+
+    public function setAuthor( string $author): void {
+        $this->author = $author;
+    }
+
+    public function setPrice(float $price): void {
+        $this->price = $price;
+    }
+
+    public function setPublisher(string $publisher): void {
+        $this->publisher = $publisher;
+    }
+    
+    public function setIsbn(string $isbn): void {
+        $this->isbn = $isbn;
+    }      
+}
+```
+If you find books boring, use games, films or whatever, it doesn't matter.
+
+So what's all this?
+
+## Describing a class
+
+First of all, we define our new `Book` class with the keyword `class`. Inside our class, we define the properties of this class, like the title, the author, the publisher etc. Of course, this book can have many more properties, but we keep it basic.
+
+### Properties and methods
+
+Class properties (variables and constants) and also the methods can have different states of visibility: private, protected or public. Properties of a class should be private or protected to keep control over what can be assigned to a property and what not.
```

Comment:
I'd write something about the difference between `private` and `protected` here as it's otherwise confusing:

> Both private and protected properties cannot be accessed from the outside of the class, private properties are even more limited in access. In most cases, protected properties are fine. You can read more about the difference below.